### DISCUSSION TITLE
why using curl as client can not call hello function in example-service server

### DISCUSSION THREAD

**Discussion by tomarock** [google/tarpc#488] [Q&A] [answered]
I using command: **curl -X POST -H "Content-Type: application/json" -d '{"jsonrpc":"2.0","method":"hello","params":["YourName"],"id":1}' http://localhost:8080** for test but it dose not return response

**Comment by tikue** [comment]
Hi! Can you share your server code?

**Reply by tomarock** [reply]
Hi tikue! 
I run example in this repo: https://github.com/google/tarpc/tree/master/example-service

**Reply by tikue** [reply, marked as answer]
Ah ok, try using `RUST_LOG=debug` to get more verbose logging. But I think the problem is that the `curl` command you used sends an HTTP request that looks like this:

```
POST / HTTP/1.1
Host: localhost:8888
User-Agent: curl/8.7.1
Accept: */*
Content-Type: application/json
Content-Length: 63

{"jsonrpc":"2.0","method":"hello","params":["YourName"],"id":1}
```

But the server binary in example-service does not use HTTP at all. The request you need to send looks something like this:

```
{"Request":{"context":{"deadline":{"secs":9,"nanos":999699542},"trace_context":{"trace_id":[8,170,113,216,215,155,43,44,255,165,13,120,158,242,13,97],"span_id":7845148348690344672,"sampling_decision":"Unampled"}},"id":0,"message":{"Hello":{"name":"YourName"}}}}
```

You can test the expected payload by listening on a TCP port and printing out the received stream (`nc -l localhost 8888` or something) and then using `cargo run --bin client` to send a request to the port.
